pkg/plugins/message/marshaler_json.go

```diff
@@ -1,27 +1,38 @@
 package message
 
 import (
-	"encoding/json"
+	cloudevents "github.com/cloudevents/sdk-go"
 )
 
+// Marshaler ...
+type Marshaler interface {
+	Marshal(interface{}) (cloudevents.Event, error)
+	Unmarshal(cloudevents.Event, interface{}) error
+}
+
 // JSONMarshaler ...
 type JSONMarshaler struct {
 	NewUUID func() string
 }
 
 // Marshal ...
-func (m JSONMarshaler) Marshal(v interface{}) (*Message, error) {
-	b, err := json.Marshal(v)
-	if err != nil {
-		return nil, err
-	}
+func (m JSONMarshaler) Marshal(v interface{}) (cloudevents.Event, error) {
+	// this is converting to a cloud event
+	e := cloudevents.NewEvent()
 
-	msg := New(m.NewUUID(), b)
+	e.SetID(m.NewUUID())
+	e.SetType("us.andersnormal.autobot.message")
+	e.SetSource("github.com/andersnormal/autobot/pkg/plugins/message")
```

Comment:
neat 👍

Reply:
curious as to how you think we would use this information - just for audit i assume?

Reply:
Yeah, auditing, and maybe provide a definition later on.